tests/agent_unittests/test_package_version_utils.py

```diff
@@ -150,3 +150,21 @@ def test_version_caching(monkeypatch):
     del sys.modules["mymodule"]
     version = get_package_version("mymodule")
     assert version not in NULL_VERSIONS, version
+
+
+@SKIP_IF_NOT_IMPORTLIB_METADATA
+def test_version_as_class_property(monkeypatch):
+    # There is no file/module here, so we monkeypatch
+    # pytest instead for our purposes
+    class FakeModule:
+        @property
+        def version(self):
+            return "1.2.3"
+
+    monkeypatch.setattr(pytest, "version", FakeModule.version, raising=False)
+    monkeypatch.setattr(
+        sys.modules["importlib"].metadata, "version", lambda x: "1.2.3", raising=False  # pylint: disable=E1101
+    )
+
+    version = get_package_version("pytest")
+    assert version == "1.2.3"
```

Comment:
To go with the suggestion above, can we get a test that does almost this exact thing but instead sets `version_tuple` as an actual tuple while `version` is a property?